src/lib/log-forwarding.js

```diff
@@ -0,0 +1,253 @@
+    rtLib.utils.checkOpenWhiskCredentials({ ow: this.aioConfig.runtime })
+    this.logForwarding = await getRTLogForwarding(this.aioConfig.runtime)
+    return this
+  }
+
+  getLocalConfig () {
+    const config = this.aioConfig.project.workspace.log_forwarding
+    try {
+      return convertToConfigObject(config)
+    } catch (e) {
+      throw new Error('Incorrect local log forwarding configuration. ' + e.message)
+    }
+  }
+
+  getLocalConfigWithSecrets () {
+    const config = this.getLocalConfig()
+    const destination = config.getDestination()
+    const settings = config.getSettings()
+    if (config.isDefined()) {
+      const destinationSettings = this.logForwarding.getDestinationSettings(destination)
+      const missingSecrets = []
+      destinationSettings.forEach(e => {
+        if (e.type === SECRET_FIELD_TYPE) {
+          const secretVarName = getSecretVarName(destination, e.name)
+          if (process.env[secretVarName] !== undefined) {
+            settings[e.name] = process.env[secretVarName]
+          } else {
+            missingSecrets.push(secretVarName)
+          }
+        }
+      })
+      if (missingSecrets.length > 0) {
+        throw new Error('Required secrets are missing in environment variables: ' + missingSecrets.join(', ') + '. ' +
+          'Make sure these variables are set in .env file')
+      }
+    }
+    return new LogForwardingConfig(destination, settings)
+  }
+
+  async getServerConfig () {
+    try {
+      return convertToConfigObject(await this.logForwarding.get())
+    } catch (e) {
+      throw new Error('Incorrect log forwarding configuration on server. ' + e.message)
+    }
+  }
+
+  getSupportedDestinations () {
+    return this.logForwarding.getSupportedDestinations()
+  }
+
+  getSettingsConfig (destination) {
+    return this.logForwarding.getDestinationSettings(destination)
+  }
+
+  async updateLocalConfig (lfConfig) {
+    const destination = lfConfig.getDestination()
+    const destinationSettings = this.logForwarding.getDestinationSettings(destination)
+    const projectConfig = {
+      project: this.aioConfig.project
+    }
+
+    const nonSecretSettings = {}
+    const secretSettings = {}
+
+    const settings = lfConfig.getSettings()
+    Object.keys(settings).forEach(k => {
+      const destFieldSettings = destinationSettings.find(i => i.name === k)
+      if (destFieldSettings.type === SECRET_FIELD_TYPE) {
+        secretSettings[getSecretVarName(destination, k)] = settings[k]
+      } else {
+        nonSecretSettings[k] = settings[k]
+      }
+    })
+
+    projectConfig.project.workspace.log_forwarding = {
+      [destination]: nonSecretSettings
+    }
+    const interactive = false
+    const merge = true
+    await writeAio(projectConfig, process.cwd(), { interactive, merge })
+    await writeEnv({}, process.cwd(), { interactive, merge }, secretSettings)
+  }
+
+  isLocalConfigChanged () {
+    if (fs.pathExistsSync(process.cwd() + '/tmp/' + CHECKSUM_FILE)) {
```

Comment:
- Use the dist/ folder for this, which is part of aioConfig
- this code is not windows friendly either
- process.cwd() is not needed
- use path.join instead of adding path strings yourself

Reply:
Updated